Public/Src/Tools/BuildXL.Tools.CredentialProvider/CredentialProviderGenerator.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace BuildXL.Tools.CredentialProvider
+{
+    class CredentialProviderGenerator
+    {
+        private static Dictionary<string, string> ValidPackageSources = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+            { "https://pkgs.dev.azure.com/1essharedassets/_packaging/BuildXL/nuget/v3/index.json", "1ESSHAREDASSETS_BUILDXL_FEED_PAT" },
+            { "https://pkgs.dev.azure.com/cloudbuild/_packaging/BuildXL.Selfhost/nuget/v3/index.json", "CLOUDBUILD_BUILDXL_SELFHOST_FEED_PAT" }
+        };
+
+        public static int Main(string[] args)
+        {
+            string uri = GetUri(args);

```

Comment:
if this program takes exactly 2 arguments, namely `-uri <some-uri>`, then why search through `args` instead of just asserting that
  - `args.Length == 2;`
  - `args[0] == "-uri";`
  - `!string.IsNullOrEmpty(args[1]);`

Reply:
Those parameters are passed by nuget.exe and it also passes some logging flags, etc. 
They may also add new flags in the future so it's safer to dynamically check it.